cpp/daal/src/data_management/roc_auc_score.cpp

```diff
@@ -0,0 +1,167 @@
+/** file roc_auc_score.cpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "data_management/data/internal/roc_auc_score.h"
+#include "data_management/data/numeric_table.h"
+#include "data_management/data/data_dictionary.h"
+#include "services/env_detect.h"
+#include "src/externals/service_dispatch.h"
+#include "src/services/service_data_utils.h"
+#include "src/services/service_utils.h"
+#include "src/services/service_defines.h"
+#include "src/threading/threading.h"
+#include "src/data_management/service_numeric_table.h"
+#include "data_management/features/defines.h"
+#include "src/algorithms/service_error_handling.h"
+#include "src/externals/service_rng.h"
+#include "src/externals/service_rng_mkl.h"
+
+namespace daal
+{
+namespace data_management
+{
+namespace internal
+{
+/*
+template <typename DataType>
+struct Sum 
+{
+    double value;
+
+    Sum() : value(0) {}
+    Sum(Sum& s, tbb::split) : value(0) {}
+
+    void operator()(const tbb::blocked_range<typename std::vector<DataType>::iterator>& r) {
+        value = std::accumulate(r.begin(), r.end(), value);
+    }
+
+    void join(Sum& rhs) { value += rhs.value; }
+};
+*/
+template <typename DataType, daal::CpuType cpu>
+void calculateRankDataImpl(DataType * predictedRank, NumericTablePtr & prediction_numpy, const int& size)
+{
+    data_management::BlockDescriptor<DataType> numpyBlock;
+    prediction_numpy->getBlockOfRows(0, 1, data_management::readOnly, numpyBlock);
+    DataType * numpyPtr = numpyBlock.getBlockPtr();
+
+    std::pair<DataType, size_t>* v_sort = new std::pair<DataType, size_t>[size];
```

Comment:
Is it allowed to use entities from "std::" namespace in old interfaces?

Reply:
pls watch comment from Smirnov Egor, I will remove std